internal/testing/cluster_test.go

```diff
@@ -1906,3 +1888,96 @@ func TestUploadPacking(t *testing.T) {
 		t.Errorf("expected totalUploadedSize of %v, got %v", totalRedundantSize, os.TotalUploadedSize)
 	}
 }
+
+func TestWallet(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	cluster, err := newTestCluster(t.TempDir(), newTestLoggerCustom(zapcore.DebugLevel))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := cluster.Shutdown(context.Background()); err != nil {
+			t.Fatal(err)
+		}
+	}()
+	b := cluster.Bus
+
+	// Check wallet info is sane after startup.
+	initialInfo, err := b.Wallet(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if initialInfo.ScanHeight == 0 {
+		t.Fatal("wallet scan height should not be 0")
+	}
+	if initialInfo.Confirmed.IsZero() {
+		t.Fatal("wallet confirmed balance should not be zero")
+	}
+	if !initialInfo.Spendable.Equals(initialInfo.Confirmed) {
+		t.Fatal("wallet spendable balance should match confirmed")
+	}
+	if !initialInfo.Unconfirmed.IsZero() {
+		t.Fatal("wallet unconfirmed balance should be zero")
+	}
+	if initialInfo.Address == (types.Address{}) {
+		t.Fatal("wallet address should be set")
+	}
+
+	// Send 1 SC to an address outside our wallet. We manually do this to be in
+	// control of the miner fees.
+	sendAmt := types.HastingsPerSiacoin
+	minerFee := types.NewCurrency64(1)
+	txn := types.Transaction{
+		SiacoinOutputs: []types.SiacoinOutput{
+			{Value: sendAmt, Address: types.VoidAddress},
+		},
+		MinerFees: []types.Currency{minerFee},
+	}
+	toSign, parents, err := b.WalletFund(context.Background(), &txn, txn.SiacoinOutputs[0].Value)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = b.WalletSign(context.Background(), &txn, toSign, types.CoveredFields{WholeTransaction: true})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := b.BroadcastTransaction(context.Background(), append(parents, txn)); err != nil {
+		t.Fatal(err)
+	}
+
+	// The wallet should still have the same confirmed balance, a lower
+	// spendable balance and a greater unconfirmed balance.
+	var info api.WalletResponse
+	err = Retry(600, 100*time.Millisecond, func() error {
+		info, err = b.Wallet(context.Background())
+		if err != nil {
+			return err
+		}
+		if !info.Confirmed.Equals(initialInfo.Confirmed) {
+			return fmt.Errorf("wallet confirmed balance should not be zero: %v %v", info.Confirmed, initialInfo.Confirmed)
+		}
+		if info.Spendable.Cmp(initialInfo.Spendable) >= 0 {
+			return fmt.Errorf("wallet spendable balance should be lower than before: %v %v", info.Spendable, initialInfo.Spendable)
+		}
+		if info.Unconfirmed.Cmp(initialInfo.Unconfirmed) < 0 {
+			return fmt.Errorf("wallet unconfirmed balance should be greater than before: %v %v", info.Unconfirmed, initialInfo.Unconfirmed)
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// The diffs of the spendable balance and unconfirmed balance should add up
+	// to the amount of money sent as well as the miner fees used.
+	spendableDiff := initialInfo.Spendable.Sub(info.Spendable)
```

Comment:
`nit` if we assert exact amounts here we might as well assert them in the `Retry` above instead of using greater than, lower than?